fakeaudit/utils.go

```diff
@@ -14,51 +15,97 @@ import (
 
 var absPath, _ = filepath.Abs("../")
 
-func ReadConfigFiles(filename string) runtime.Object {
+func ReadConfigFiles(filename string) (decoded []runtime.Object, err error) {
 	buf, err := ioutil.ReadFile(filename)
 
 	if err != nil {
 		log.Error("File not found")
 		panic(err)
 	}
+	buf_slice := bytes.Split(buf, []byte("---"))
 
 	decoder := scheme.Codecs.UniversalDeserializer()
-	obj, _, err := decoder.Decode(buf, nil, nil)
 
-	if err != nil {
-		log.Errorf("Could not decode the given yaml: %s\n%s", string(buf), err)
+	for _, b := range buf_slice {
+		obj, _, err2 := decoder.Decode(b, nil, nil)
+		err = err2
+		decoded = append(decoded, obj)
 	}
-
-	return obj
+	return
 }
 
-func getDeployment(filename string) (deployment *v1beta1.Deployment) {
-	obj := ReadConfigFiles(filename)
-	deployment = obj.(*v1beta1.Deployment)
+func getDeployment(filename string) (deployment *v1beta1.Deployment, err error) {
+	obj_slice, err := ReadConfigFiles(filename)
+	if err != nil {
+		return
+	}
+	for _, obj := range obj_slice {
+		switch resource := obj.(type) {
+		case *v1beta1.Deployment:
```

Comment:
Might also want to add a default to all these switch cases and add some sort of error like `Unsupported param`

Reply:
Or since it's a single case why not if/else

Reply:
Interesting I think all these functions like `getDeployment` etc can go away and we can just work with `getKubeResource` from cmd/utils.go and we can transfer that function in fakeaudit/utils.go

Reply:
Yay! 🎉  lot of dead code removal